### ISSUE TITLE
Bump version inside VersionInfo directly

### ISSUE THREAD

**Issue by tomschr** [k-bx/python-semver#139]
## Situation

Hope I don't miss anything, but reading the source code didn't give a solution. It seems to me, dealing with version numbers and raising parts of it can actually quite tricky.

For example, let's assume you've created a version like this:

```python
ver = semver.parse_version_info("3.4.5")
```

How can you raise the major part? How the minor part? The `ver` variable is of type `VersionInfo`. You can't directly use `semver.bump_major` or `semver.bump_minor` as they accept strings only. Nor does the class provide any methods. However,  you *could* do something like that:

```python
semver.bump_major(str(ver))
```

This works, but your `VersionInfo` type is converted into a string. Not good if you want to keep the type. If you want it back, you need to write:

```python
semver.parse_version_info(semver.bump_major(str(ver)))
```

Not very pythonic. :wink: 

## Possible Solution(s)

From what I can see, we have a good class, but we can't raise parts individually. I think, this is gap. I could think of these solutions:

1. Allow `semver.bump_major` and the other bump functions allow a `VersionInfo` type.
2. Extend the `VersionInfo` class with additional methods like `inc_major()`, `inc_minor()` etc.
3. Something else?

At the moment, I think number 2 can be a possible solution. Theoretically, you could name them also `bump_major`, but that could confuse people.

The above could be rewritten like that:

```python
ver.inc_major()
```

If we go with this, there are two possible ways how this function could work:

1. Return the raised version of `ver` but leave `ver` itself untouched.
2. Return nothing (=None) and raise the part directly in `ver`.

Solution 1 could be interesting as it would allow to write something like this:

```python
new_ver = ver.inc_major().inc_minor()
```

What do you think?

**Comment by ppkt** [comment]
@tomschr you're right, currently we don't have such feature :)

Personally, I like OO approach here and I think adding these functions would be really nice. Regarding your proposal - I like options 2 and 1 respectively (but I'd like to stick with `bump_minor` to match name with functions working on strings). As alternative, we can add new function as class method to avoid confusion regarding whether function returns something or modify object itself:
`new_ver = VersionInfo.bump_major(ver)`

**Comment by tomschr** [comment]
> @tomschr you're right, currently we don't have such feature :)

Ahh good, that means, I didn't overlook anything. :wink: 


> Personally, I like OO approach here and I think adding these functions would be really nice. 

Mee too.

> Regarding your proposal - I like options 2 and 1 respectively (but I'd like to stick with `bump_minor` to match name with functions working on strings). As alternative, we can add new function as class method to avoid confusion regarding whether function returns something or modify object itself:
> `new_ver = VersionInfo.bump_major(ver)`

Sure, we can use the same name here. Give me some time so I can up with a draft pull request.

Thanks for your answer! :+1:

**Comment by tomschr** [comment]
@ppkt  Karol, I found something which we should clarify.

> [...] As alternative, we can add new function as class method to avoid confusion regarding whether function returns something or modify object itself:
> `new_ver = VersionInfo.bump_major(ver)`

I'm not sure if it should be a class method. **If** it would be a class method and you want to bump major and minor, you have to write this:

```python
new_ver = VersionInfo.bump_major(ver)
new_ver = VersionInfo.bump_minor(new_ver)
```

If I'm not mistaken, you can't concatenate the methods like this:

```python
new_ver = ver.bump_major().bump_minor()
```

Should we allow/forbid such use case? :thinking:

**Comment by ppkt** [comment]
@tomschr it's an interesting use case - I've never thought about it. I think, from user point of view, better approach is to use regular method and return updated object (as you suggested) and forget about class method :)

**Comment by tomschr** [comment]
Uups, this was already fixed in commit a598059, but wasn't closed automatically.
